marshmallow_recipe/bake.py

```diff
@@ -240,13 +237,21 @@ def get_field_for(
             )
 
         if origin is Annotated:
+            underlying_type, *annotations = arguments
+            annotated_metadata = next((x for x in annotations if isinstance(x, Metadata)), Metadata())
+            metadata = dict(metadata, **annotated_metadata)
+
             return get_field_for(
-                arguments[0],
+                underlying_type,
                 metadata=metadata,
                 naming_case=naming_case,
                 none_value_handling=none_value_handling,
             )
 
+    field_factory = _SIMPLE_TYPE_FIELD_FACTORIES.get(type)
```

Comment:
We had to move it before parsing Annotated, because Annotated[T, dict] is not hashable.